File: tool/src/types/supplySource.ts
export interface SelectRequest {
  productName: string;
  productOption?: string;
  imageUrls?: string[];
  quantity: number;
}

export interface SelectResponse {
  success: boolean;
  data?: {
    supplier: 'TAOBAO';
    productName: string;
    productOption: string;
    productUrl: string;
    quantity: number;
    unitPrice: number;
  };
  error?: string;
}

File: core/src/main/kotlin/com/ai/seller/domain/SupplySource.kt
package com.ai.seller.domain

data class SupplySource(
    val supplier: Supplier,
    val product: Product,
) {
    enum class Supplier {
        TAOBAO;
    }

    data class Product(
        val productName: String,
        val productOption: String,
        val productUrl: String,
        val quantity: Int,
        val unitPrice: Int,
    )
}

interface SupplySourceSelector {
    fun select(product: Order.Product): SupplySource
}
